syscalls.go

```diff
@@ -201,10 +209,19 @@ var objNameAllowsDot = internal.NewFeatureTest("dot in object names", func() err
 		MapName:    sys.NewObjName(".test"),
 	}
 
+	// Tolerate EPERM, otherwise MapSpec.Name has its dots removed when run by
```

Comment:
@ti-mo Just found this as I was rebasing other work. I think the fact that the elf reader changes based on the host system is the problem here, not that the test doesn't account for not having permissions. I'll see whether I can untangle my mess.